SmartPlaces.Facilities/lib/IngestionManager/src/RedisTwinMappingIndexer.cs

```diff
@@ -34,19 +34,19 @@ public RedisTwinMappingIndexer(IDistributedCache cache)
                                                         (retryAttempt) => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)) + TimeSpan.FromMilliseconds(jitter.Next(0, 100)));
         }
 
-        public async Task<TwinMap?> GetTwinIndexAsync(string sourceId)
+        public async Task<TwinMapEntry?> GetTwinIndexAsync(string sourceId)
         {
             return await redisRetryPolicy.ExecuteAsync(async () =>
             {
                 var cacheValue = await cache.GetStringAsync(sourceId);
-                var twinMap = JsonSerializer.Deserialize<TwinMap>(cacheValue);
-                return twinMap;
+                var mapEntry = JsonSerializer.Deserialize<TwinMapEntry>(cacheValue);
```

Comment:
Should this PR introduce the null check or is that coming in a different PR?

Reply:
Good catch, I actually missed sorting this out. Will fix and resubmit. Thanks!